nss-day-cohort-29/nutshell-divine-madness
# Messages functionality complete

# Description

Fully implements the functionality required in the messages issue.

## Type of change

Please delete options that are not relevant.



# Testing Instructions for Change Made

- Checkout branch rr-messagesSecond
- run grunt
- Page will load and display the stored messages in order by Date/Time
     - Test sorting by editing time on the entires in the json and refereshing
- There is an input at the bottom accompanied by a submit button for creating new messages
     - Test by entering a new message and you will see the message list refresh with the new message at the end
- Each message by the user who is logged in will have an "Edit" button next to it
- When clicked, a form to edit the message will show up with the text from the database prepopulted along with a submit button for the edit
     - Change the text and hit submit and you will see the change in the json as well as in the displayed messages
- Editing a message does not change the original time stamp, thus preserving the conversational order of the messages

# Checklist:

- [x ] My code follows the style guidelines of this project
- [x ] I have performed a self-review of my own code
- [x ] I have commented my code, particularly in hard-to-understand areas
- [x ] My changes generate no new warnings
- [x ] I have added test instructions that prove my fix is effective or that my feature works
